Code/SG Scottish manufacturer analysis Sep24.R

```diff
@@ -0,0 +1,106 @@
+####################################################
+#SG ANALYSIS#
+#What proportion of total HFSS sales in 2023 (by price and volume) were made by Scottish manufacturers  #
+#Written by Elaine Tod 3rd September 2024
+####################################################
+
+#Set libraries and working directories
+library(tidyverse) #lots of functions
+library(arrow) # for efficient file saving and reading
+library(tidyr)
+library(janitor)
+library(dplyr)
+library(readr)
+library(data.table) # For 'fread' function to read in CSVs efficiently                                        # Install xlsx R package                                                # Load xlsx R package to RStudio
+setwd("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/")
+
+
+###########################################################################
+#PART 1:CREATE SG DATASET 2023 (PURCHASES WITH MANUFACTURER AND HFSS INFO)#
+###########################################################################
+
+HFSSFINAL23 <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/HFSSFINAL23.parquet")#Open linked data file for 2023
+prodnames <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/npm_hfss_prod_names.parquet")
+
+#Kantar_regcats <- fread("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/PHS HFSS Itemisation Breakdown amended.csv")
+#IGNORE - KEPT IN CASE NEEDED THEN WILL DELETE
+#files <- list.files(path = "2023 Purchase Data/product master", full.names = TRUE)#Create combined RFfile
+#dataframes <- lapply(files, function(f){
+  #fread(f, skip = 1, header = TRUE) %>%
+    #select(any_of(c("Product", "Product Desc", "Manufacturer", "Brand", "RST 4 Market", "RST 4 Sub Market", "RST 4 Extended", "Low Sugar/Calorie/Fat")))
+#}
+#)
+
+#rf_combined_SG <- do.call(bind_rows, dataframes)
+
+  #write_parquet(rf_combined_SG, "rf_combined_SG.parquet")
+
+#rf_combined_SG <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/rf_combined_SG.parquet")#Link manufacturer data to HFSSFINAL22_23 data file using a version of Catriona's 'rf_markets_combined.parquet file - saved separately.
+
+
+#QA - remove 47 products with missing manufacturer info
+#rf_combined_SG <- rf_combined_SG[-which(rf_combined_SG$Manufacturer == ""), ]
+
+
+#18/09/24 - START HERE
+
+#Get Catriona's file with HFSS categories
+prodnames <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/npm_hfss_prod_names.parquet")
+
+#Merge master purchase dataset for 2023 with HFSS regulation category file
+SG_dataset <- dplyr::left_join(HFSSFINAL23,prodnames, by=c("prodcode" = "PRODUCT"))
```

Comment:
You can avoid creating duplicate columns and then deleting them by only selecting the columns you're interested in from the prodnames dataframe. You could create a new dataframe, e.g.

prodnames_to_join <- prodnames %>%
select(PRODUCT, hfss_categories)

or do it as part of the join, as below:

```suggestion
SG_dataset <- dplyr::left_join(HFSSFINAL23,prodnames %>% select(PRODUCT, hfss_category), by=c("prodcode" = "PRODUCT"))
```